oxenc/bt_producer.h

```diff
@@ -141,21 +157,59 @@ class bt_list_producer {
     /// beyond the end of the buffer an exception is raised.  Note that this will happen during
     /// construction if the given buffer is not large enough to contain the `le` encoding of an
     /// empty list.
-    bt_list_producer(char* begin, char* end) : bt_list_producer{begin, end, "l"sv} {}
+    bt_list_producer(char* begin, char* end) : bt_list_producer{begin, end, 'l'} {}
 
     /// Constructs a list producer that writes into the range [begin, begin+size).  If a write would
     /// go beyond the end of the buffer an exception is raised.
-    bt_list_producer(char* begin, size_t len) : bt_list_producer{begin, begin + len, "l"sv} {}
+    bt_list_producer(char* begin, size_t len) : bt_list_producer{begin, begin + len, 'l'} {}
+
+    /// Constructs a list producer that writes to an internal, expandable string.  `reserve` can
+    /// be passed a non-zero value to reserve an initial size in the std::string.
+    explicit bt_list_producer(size_t reserve = 0) : bt_list_producer{'l', reserve} {}
 
     ~bt_list_producer();
 
     /// Returns a string_view into the currently serialized data buffer.  Note that the returned
     /// view includes the `e` list end serialization markers which will be overwritten if the list
     /// (or an active sublist/subdict) is appended to.
-    std::string_view view() const { return {from, static_cast<size_t>(to - from)}; }
+    ///
+    /// (If you just want a copy of the string, use `view()` instead).
```

Comment:
this seems self-referential and incorrect, perhaps it should be "use `str()` instead"?